wcfsetup/install/files/lib/system/user/rank/command/SaveContent.class.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace wcf\system\user\rank\command;
+
+use wcf\system\WCF;
+
+/**
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ */
+final class SaveContent
+{
+    public function __construct(
+        public readonly int $rankID,
+        /** @var array<int, string> */
+        public readonly array $titles
+    ) {
+    }
+
+    public function __invoke(): void
+    {
+        if ($this->titles === []) {
+            return;
+        }
+
+        WCF::getDB()->beginTransaction();
```

Comment:
There is no need for this transaction. Modern MySQL versions actually keep the prepared statement around thus this adds another moving part without making much of a difference. We don’t care if this operation is actually atomic.